aws-sns-topicinlinepolicy/src/test/java/software/amazon/sns/topicinlinepolicy/CreateHandlerTest.java

```diff
@@ -0,0 +1,605 @@
+                .build();
+
+
+        final ResourceHandlerRequest<ResourceModel> request = ResourceHandlerRequest.<ResourceModel> builder()
+                .desiredResourceState(model)
+                .build();
+
+        assertThrows(CfnInvalidRequestException.class, () -> handler.handleRequest(proxy, request, new CallbackContext(), proxyClient, logger));
+    }
+
+    @Test
+    public void handleRequest_Failure_InvalidParameterException() {
+        when(proxyClient.client().getTopicAttributes(any(GetTopicAttributesRequest.class)))
+                .thenReturn(GetTopicAttributesResponse.builder()
+                        .attributes(attributes)
+                        .build());
+
+        String topic = "arn:aws:sns:us-east-1:123456789012:sns-topic-name";
+
+        Map<String, Object> policyDocument = getSNSPolicy();
+
+        final ResourceModel model = ResourceModel.builder()
+                .topicArn(topic)
+                .policyDocument(policyDocument)
+                .build();
+
+        final ResourceHandlerRequest<ResourceModel> request = ResourceHandlerRequest.<ResourceModel> builder()
+                .desiredResourceState(model)
+                .build();
+
+        when(proxyClient.client().setTopicAttributes(any(SetTopicAttributesRequest.class)))
+                        .thenThrow(InvalidParameterException.class);
+        final ProgressEvent<ResourceModel, CallbackContext> response = handler.handleRequest(proxy, request,
+                new CallbackContext(), proxyClient, logger);
+        assertThat(response.getStatus()).isEqualTo(OperationStatus.FAILED);
+        assertThat(response.getErrorCode()).isEqualTo(HandlerErrorCode.InternalFailure);
+
+        verify(proxyClient.client(), times(1)).setTopicAttributes(any(SetTopicAttributesRequest.class));
+    }
+
+    @Test
+    public void handleRequest_INPROCESS_InvalidParameterExceptionWithWrongPrincipal() {
+        when(proxyClient.client().getTopicAttributes(any(GetTopicAttributesRequest.class)))
+                .thenReturn(GetTopicAttributesResponse.builder()
+                        .attributes(attributes)
+                        .build());
+
+        String topic = "arn:aws:sns:us-east-1:123456789012:sns-topic-name";
+
+        Map<String, Object> policyDocument = getSNSPolicy();
+
+        final ResourceModel model = ResourceModel.builder()
+                .topicArn(topic)
+                .policyDocument(policyDocument)
+                .build();
+
+        final ResourceHandlerRequest<ResourceModel> request = ResourceHandlerRequest.<ResourceModel> builder()
+                .desiredResourceState(model)
+                .build();
+
+        InvalidParameterException exception = InvalidParameterException.builder()
+                .awsErrorDetails(AwsErrorDetails.builder()
+                        .errorCode("InvalidParameter")
+                        .errorMessage("Invalid parameter: Policy Error: PrincipalNotFound").build())
```

Comment:
How about making `Invalid parameter: Policy Error: PrincipalNotFound` as static, and referring in both code and tests?